cmd/snap/cmd_snap_op.go

```diff
@@ -369,9 +369,12 @@ func showDone(cli *client.Client, names []string, op string, opts *client.SnapOp
 				msg = fmt.Sprintf(i18n.G("%q left the cohort"), snap.Name)
 			}
 			fmt.Fprintln(Stdout, msg)
+			// go to next snap, skipping the temporary forward logic below for "switch" op.
+			continue
 		default:
 			fmt.Fprintf(Stdout, "internal error: unknown op %q", op)
 		}
+		// XXX: this makes sense only for certain operations (refresh?), but not for "switch"
```

Comment:
Hmm. It only makes sense for install and refresh... right now for example `revert` could also trigger it and there's no "forwarding" going on there either.
As we're wanting to refactor it asap, probably before default channels, I think it makes sense to keep this particular change minimal. As such, adding a check for `op == "install" || op == "refresh"` before we check whether we need to print the notice would be best, even if introducing helper functions would be cleaner.

Reply:
Sounds good to me, so lets go with that!

Reply:
Thanks for the suggestions, done, although note, there was a unit test for revert that expected the "fowarding" message. I assume the test was wrong then @chipaca ?

Reply:
yeah, I think it was wrong. From a particular point of view it was technically correct, but there's no point in bringing that to the attention of the user at that time as it has nothing to do with what snapd just did on their behalf. Also `revert` doesn't check the store so there's no way of knowing if the "forwarding" is still a thing at that point. So, yeah, no.

Reply:
Good, thanks for clarification.